include/deal.II/lac/arpack_solver.h

```diff
@@ -343,6 +357,52 @@ class ArpackSolver : public Subscriptor
         std::vector<VectorType> &          eigenvectors,
         const unsigned int                 n_eigenvalues = 0);
 
+  /**
+   * Solve the eigensprectrum problem $A x=\lambda x$ by calling the
+   * <code>dsaupd</code> and <code>dseupd</code> or <code>dnaupd</code> and
+   * <code>dneupd</code> functions of ARPACK.
+   *
+   * The function returns a vector of eigenvalues of length <i>n</i> and a
+   * vector of eigenvectors of length <i>n</i> in the symmetric case
+   * and of length <i>n+1</i> in the non-symmetric case. In the symmetric case
+   * all eigenvectors are real. In the non-symmetric case complex eigenvalues
+   * always occur as complex conjugate pairs. Therefore the eigenvector for an
+   * eigenvalue with nonzero complex part is stored by putting the real and
+   * the imaginary parts in consecutive real-valued vectors. The eigenvector
+   * of the complex conjugate eigenvalue does not need to be stored, since it
+   * is just the complex conjugate of the stored eigenvector. Thus, if the last
+   * n-th eigenvalue has a nonzero imaginary part, Arpack needs in total n+1
+   * real-valued vectors to store real and imaginary parts of the eigenvectors.
+   *
+   * @param A The operator for which we want to compute eigenvalues.
+   *
+   * @param eigenvalues is a vector of complex numbers in which the
+   * eigenvalues are returned.
+   *
+   * @param eigenvectors is a <b>real</b> vector of eigenvectors, containing
+   * the real parts of all eigenvectors and the imaginary parts of the
+   * eigenvectors corresponding to complex conjugate eigenvalue pairs.
+   * Therefore, its length should be <i>n</i> in the symmetric case and
+   * <i>n+1</i> in the non-symmetric case. In the non-symmetric case the storage
+   * scheme leads for example to the following pattern. Suppose that the first
+   * two eigenvalues are real and the third and fourth are a complex conjugate
+   * pair. Asking for three eigenpairs results in <i>[real(v1),real(v2),
+   * real(v3),imag(v3)]</i>. Note that we get the same pattern if we ask for
+   * four eigenpairs in this example, since the fourth eigenvector is simply the
+   * complex conjugate of the third one.
+   *
+   * @param n_eigenvalues The purpose of this parameter is not clear, but it
+   * is safe to set it to the size of <code>eigenvalues</code> or greater.
+   * Leave it at its default zero, which will be reset to the size of
```

Comment:
-> If you leave it at zero, then the number of eigenvalues to be computed simply equals the size of @p eigenvalues.

Though if it's unclear what it does, why not just omit the argument?

Reply:
I just used the same interface than what we have right now.